web/handlers.go

```diff
@@ -299,6 +299,12 @@ func (h *Handler) checkCSRFToken(c *Context, r *http.Request, token string, toke
 
 	if csrfCheckNeeded {
 		csrfHeader := r.Header.Get(model.HEADER_CSRF_TOKEN)
+		if csrfHeader == "" && *c.App.Config().ServiceSettings.EnableCWSTokenLogin {
```

Comment:
This concerns me a bit @jupenur . I've added it because in case the user from another domain tries to enter again using the same CWS route (form and so on) gets a session error because is not able to fetch de CSRF token from the cookie. Do you see anything weird/wrong?

Reply:
Yes, this is effectively disables the anti-CSRF measures for instances that set `EnableCWSTokenLogin`. The solution is to split the autlogin into its own REST endpoint and use `api.ApiHandlerTrustRequester` for the handler.